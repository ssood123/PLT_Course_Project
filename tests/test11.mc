float f;

function void main()
{
  f = 88.88;
  printf(f);
  main1();
  printf(f);
}

function void main1()
{
  f = 90.91;
}
