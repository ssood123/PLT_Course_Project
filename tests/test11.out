; ModuleID = 'integraph'
source_filename = "integraph"

@f = global double 0.000000e+00
@fmt = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.1 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.2 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1
@fmt.3 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@fmt.4 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1
@fmt.5 = private unnamed_addr constant [4 x i8] c"%g\0A\00", align 1

declare i32 @printf(i8*, ...)

declare i32 @printbig(i32)

define void @main1() {
entry:
  store double 9.091000e+01, double* @f, align 8
  ret void
}

define void @main() {
entry:
  store double 0x40563851EB851EB8, double* @f, align 8
  %f = load double, double* @f, align 8
  %printf = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.5, i32 0, i32 0), double %f)
  call void @main1()
  %f1 = load double, double* @f, align 8
  %printf2 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @fmt.5, i32 0, i32 0), double %f1)
  ret void
}
